site-scrapers/TrinityEMS/index.js

```diff
@@ -0,0 +1,253 @@
+const s3 = require("../../lib/s3");
+const { site } = require("./config.js");
+const { sendSlackMsg } = require("../../lib/slack");
+
+module.exports = async function GetAvailableAppointments(browser) {
+    const siteName = site.name;
+    console.log(`${siteName} starting.`);
+    const availability = await ScrapeWebsiteData(browser, site);
+    console.log(`${siteName} done.`);
+    return {
+        ...site,
+        ...availability,
+    };
+};
+
+/**
+ * Flag to indicate that the page content has been writtent to s3 because a date with
+ * availability has been found and logged.
+ */
+let pageContentSavedToS3 = false;
+
+async function ScrapeWebsiteData(browser, site) {
+    const page = await browser.newPage();
+
+    await page.goto(site.website);
+    try {
+        // Wait until Javascript has loaded availabity data
+        const classToWaitFor = ".scheduleday";
+        await waitForLoadComplete(page, classToWaitFor);
+    } catch (error) {
+        console.log(`error waiting for `);
+        return { hasAvailability: false };
```

Comment:
I think we want to return  `{ availability: {}, hasAvailability: false };`
In which case you can initialize `results` above and return this here.

Reply:
I don't think it matters if the `availability` key is present or not but it's a good idea to have it.

Reply:
Since it looks like I'm the only one who trapped for errors, I'll remove the try/catch and let it rip.